api/db/populate-attendants.js

```diff
@@ -0,0 +1,83 @@
+const { getRandomSubset, createPrompt } = require('./utils');
+
+const User = require('../models/User');
+const Event = require('../models/Event');
+const Attendant = require('../models/Attendant');
+
+let displayLogs;
+
+const createAttendant = async (attendantParams) => {
+  const attendant = new Attendant({
+    event: attendantParams.event,
+    user: attendantParams.user,
+  });
+  await attendant.save();
+
+  if (displayLogs) {
+    console.log('\n', '\x1b[0m', `New attendant created: ${attendant}`);
+  }
+};
+
+const createAttendants = async (verbose) => {
+  console.log('\n', '\x1b[0m', 'Populating attendants collection...');
+
+  let usersList = await User.distinct('_id');
+  if (usersList.length === 0) {
+    console.log(
+      '\x1b[33m',
+      'The users collection does not exist. Aborted attendants collection population.'
+    );
+    return;
+  }
+
+  let eventsList = await Event.find({
+    startDate: { $gte: new Date() },
+  });
+  if (eventsList.length === 0) {
+    console.log(
+      '\x1b[33m',
+      'The events collection does not exist. Aborted attendants collection population.'
+    );
+    return;
+  }
+
+  displayLogs = verbose;
+  let attendantsList = await Attendant.find();
+  const attendantsArray = [];
+
+  let proceed = true;
+  if (attendantsList.length !== 0) {
+    proceed = await createPrompt(
+      'Some events already contain attendees. Would you like to add attendees to events that already contain attendees?'
+    );
+  }
+
+  for (const event of eventsList) {
+    const attendees = attendantsList
+      .filter((attendee) => attendee.event.toString() == event._id.toString())
+      .map((attendee) => attendee.user.toString());
+
+    const spotsLeft = event.attendantsLimit - attendees.length;
+    if (spotsLeft <= 0 || !proceed) continue;
+
+    const potentialAttendees = usersList.filter(
+      (user) => !attendees.includes(user.toString())
+    );
+
+    const creatoreIncluded = attendees.includes(event.createdBy.toString());
+    const subset = getRandomSubset(
+      potentialAttendees,
+      creatoreIncluded ? spotsLeft : spotsLeft - 1
+    );
+    if (!creatoreIncluded) subset.unshift(event.createdBy.toString());
```

Comment:
As this is now, we could have the creator included twice `subset`, right? When `creatorIncluded` is `false`, we should the creator them from `potentialAttendees` before getting the random subset to avoid that.

Also, shouldn't this be `creatorIncluded`, without the extra `e`? And while we talk about names, `subset` could also use a better name, but feel free to leave it as is!

Reply:
Yeah, adding the `e` was a typo.